src/attachmentlistmodel.cpp

```diff
@@ -161,6 +166,24 @@ void AttachmentListModel::onAttachmentPathChanged(const QString &attachmentLocat
     }
 }
 
+void AttachmentListModel::onDirectoryChanged(const QString &path)
+{
+    for (int i = 0; i < m_attachmentsList.count(); ++i) {
+        Attachment *item = m_attachmentsList.at(i);
+        if (item->url.startsWith(QUrl::fromLocalFile(path).toString())) {
+            QFileInfo file(item->url.mid(7));
```

Comment:
Rather QUrl::toLocalFile() not to break on escape codes?

Reply:
Ah, escape codes, you're right. I've amended the commit. Thanks.